docs/themes/getting_started.rst

```diff
@@ -1,4 +1,182 @@
+
+Directory Structure
+==========================================================
+
+A theme is made of three components: the config, twig files (HTML content), and thumbnails::
+
+    config.json
+        html/
+            email.html.twig
+            form.html.twig
+            message.html.twig
+            page.html.twig
+    thumbnail.png
+    thumbnail_email.png
+    thumbnail_form.png
+    thumbnail_page.png
+
+Config File
+-----------
+
+The config file tells Mautic how to utilize the theme.
+
+.. code-block:: json
+
+    {
+      "name": "Theme Name",
+      "author": "John Doe",
+      "authorUrl": "https://john-doe-the-mautic-theme-builder.com",
+      "features": [
+        "email",
+        "form",
+        "page"
+      ],
+      "builder": ["legacy", "grapesjsbuilder"]
+    }
+
+name
+    This is the name of the theme as displayed in Mautic.
+author
+    This displays in the theme manager as credit to the author of the theme.
+authorUrl
+    URL used for the author link.
+features
+    An array of strings that tells Mautic which features the theme supports. Currently recognized values are ``email`` (email builder template), ``form`` (formatting the form page) and ``page`` (page builder template). A corresponding ``html/[feature].html.twig`` file is required for each feature supported. For example, if the theme supports ``email``, then there should be a ``html/email.html.twig`` file. See Twig Files more information on each feature.
+builder
+    Array of strings for which builder the theme supports. Currently only applies to themes that support ``page`` or ``email``. By default, themes without this will only be recognized by Mautic's legacy builder. New themes built should declare the specific builders it supports.
+
+Twig Files
+-------------------------
+
+html/message.html.twig
+^^^^^^^^^^^^^^^^^^^^^^
+
+This file is a simple message file mainly used as the landing page for when a lead unsubscribes or resubscribes to the system’s emails. But may be used by other areas so should be included in all themes.
+
+It requires echoing two variables: ``message`` and ``content``. ``message`` contains the string message such as "You have been unsubscribed..." content will either be empty or contain the HTML of a form that's been associated with the email as an unsubscribe form.
+
+.. code-block:: twig
+
+    <html>
+        <head></head>
+        <body>
+            <div>
+                <h2>{{ message|raw }}</h2>
+                {% if content is defined %}
+                <div>{{ content|raw }}</div>
+                {% endif %}
+            </div>
+        </body>
+    </html>
+
+html/email.html.twig
+^^^^^^^^^^^^^^^^^^^^^^
+
+This file defines the base template when creating a new email and should contain HTML suited for email clients.
+
+The GrapesJs builder supports the `mjml email framework <https://mjml.io/>`__.
+
+.. code-block:: html
```

Comment:
Technically, we should be using email.mjml.twig rather than email.html.twig, however the uploader in Mautic would not work properly (it required the email.html.twig) and nobody was able to fix it, so this was a hacky workaround to add support for reading MJML in the HTML file. Perhaps we should not be telling folks to do that in the docs? I'm not sure. Maybe worth discussing more widely.

Reply:
Is this the only way to get it to work at the moment? Were any other work arounds discussed? If it's the only way, then I think we have to leave it or people will not know how to build their own themes. Unless the proposal is to force people to manually install themes without going through the uploader? 

From a product perspective, something would have to tell Mautic what files are required. That should probably be the builder so maybe the validation code needs to be refactored to dispatch an event where the enabled builder can do its own validation.

Reply:
I think that it is the only way at present to use the theme uploader. We had wanted to allow the upload of themes with an email.mjml.twig file for MJML themes but nobody was available to make the necessary changes so this was the workaround. So I guess if we are not going to change that behaviour then we keep it as we have it.